From: Mike Marciniszyn <mike.marciniszyn@intel.com>
Subject: IB/rdamvt: fix cross build with rdmavt
Git-commit: 9debaaecb9070c05bbc3b5d2257d6aea416ccab6
Patch-mainline: v4.6
References: bsc#973818, fate#319242
Acked-by: Jiri Bohac <jbohac@suse.cz>

The new check routine causes a larger than supported frame size
on s390.

Changing the check routine to noinline fixes the issue.

Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/sw/rdmavt/vt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/sw/rdmavt/vt.c
+++ b/drivers/infiniband/sw/rdmavt/vt.c
@@ -383,7 +383,7 @@ static inline int check_driver_override(
 	return 1;
 }
 
-static int check_support(struct rvt_dev_info *rdi, int verb)
+static noinline int check_support(struct rvt_dev_info *rdi, int verb)
 {
 	switch (verb) {
 	case MISC:
